### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by mazar-farran**
Removes deviceId from existing frameKMs to allow them to be uploaded

- [ ] Add a link to the ticket in the PR title or add a description of work in the PR title
- [ ] dashcam firmware was loaded on a device and successfully runs
- [ ] semantic version was incremented inside `src/config/index.ts`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+      try {
+        console.log('removing deviceid');
+        const parsed = JSON.parse(contents);
+        delete parsed.deviceId;
+        res.status(200).send(JSON.stringify(parsed));
+        next();
+      } catch(e) {
+        console.log(e);
+        res.status(400).send();
+        next();
```
Comment (evanmoss) [inline src/index.ts:80]: need to `return` vs. `next()`

File: src/index.ts [hunk 2]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+    if (!filename.includes('.json')) {
+      res.status(200).send(contents);
+      next();
+    } else {
+      try {
+        console.log('removing deviceid');
+        const parsed = JSON.parse(contents);
+        delete parsed.deviceId;
+        res.status(200).send(JSON.stringify(parsed));
+        next();
```
Comment (evanmoss) [inline src/index.ts:76]: need to `return` vs. `next()`

File: src/index.ts [hunk 3]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+    let contents = '';
+    try {
+      contents = await readFile(fullPath, 'ascii');
+    } catch (e) {
+      console.log(`${fullPath} not found`);
+      next();
+    }
+
+    if (!filename.includes('.json')) {
+      res.status(200).send(contents);
```
Comment (evanmoss) [inline src/index.ts:68]: probably want to just do `next()` then `return` without sending anything on `res`

File: src/index.ts [hunk 4]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+    const fullPath = path.join(METADATA_ROOT_FOLDER, filename);
+    console.log(`reading ${fullPath}`);
+    res.setHeader("Content-Type", "application/json");
+    
+    let contents = '';
+    try {
+      contents = await readFile(fullPath, 'ascii');
+    } catch (e) {
+      console.log(`${fullPath} not found`);
+      next();
```
Comment (evanmoss) [inline src/index.ts:64]: need to return after calling `next()`

File: src/index.ts [hunk 5]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+// }
 
 export async function initAppServer(): Promise<Application> {
   const app: Application = express();
 
+  app.get('/public/metadata/:filename', async (req, res, next) => {
+    const filename = req.params.filename;
+    const fullPath = path.join(METADATA_ROOT_FOLDER, filename);
+    console.log(`reading ${fullPath}`);
+    res.setHeader("Content-Type", "application/json");
```
Comment (evanmoss) [inline src/index.ts:57]: best to change headers right before you send data through `res`

File: src/index.ts [hunk 6]
```diff
@@ -21,13 +21,71 @@ import { Instrumentation } from 'util/instrumentation';
...
+    }
+
+    if (!filename.includes('.json')) {
+      res.status(200).send(contents);
+      next();
+    } else {
+      try {
+        console.log('removing deviceid');
+        const parsed = JSON.parse(contents);
+        delete parsed.deviceId;
```
Comment (punov) [inline src/index.ts]: `parsed.bundle.deviceId`

**Review by punov — APPROVED** [review]

**Pull request merged**